test/test_utilities.jl

```diff
@@ -42,6 +42,36 @@
         @test size(y) == (2, 1)
     end
 
+    @testset "Parametric constants" begin
+        z = Constant([1.0 0.0im; 0.0 1.0])
+        @test z isa Constant{Matrix{Complex{Float64}}}
+
+        # Helper functions
+        @test Convex._size(3) == (1, 1)
+        @test Convex._sign(3) == Positive()
+        @test Convex._size([-1,1,1]) == (3, 1)
+        @test Convex._sign([-1,1,1]) == NoSign()
+        @test Convex._sign([-1,-1,-1]) == Negative()
+        @test Convex._size([0 0; 0 0]) == (2, 2)
+        @test Convex._sign([0 0; 0 0]) == Positive()
+        @test Convex._size(0+1im) == (1, 1)
+        @test Convex._sign(0+1im) == ComplexSign()
+
+        x = Constant(1, ComplexSign())
+        @test x isa Constant{Complex{Int}}
+        @test evaluate(x) === 1+0im
+        @test sign(x) == ComplexSign()
+
+        y = Constant(0+0im, Positive())
+        @test y isa Constant{Int}
+        @test evaluate(y) === 0
+        @test sign(y) == Positive()
+
+        @test_throws ArgumentError Constant(1+2im, Positive())
+        @test_throws ArgumentError Constant([-1,0], Positive())
```

Comment:
Oh, the conditions I wrote were specific to complex-valued inputs, so this doesn't error. Whoops.